Sources/App/Auth/Controllers/SpeakerController.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  File.swift
+//  
+//
+//  Created by Joe Williams on 09/04/2022.
+//
+
+import Foundation
+import Fluent
+import Vapor
+
+struct SpeakerController: RouteCollection {
+    
+    struct ImageUpload: Content {
+        var profileImage: File
+    }
+    
+    func boot(routes: RoutesBuilder) throws {
+        routes.get("create-speaker", use: onCreate)
+        let grouped = routes.grouped("api", "v1", "speaker")
+        grouped.post("create", use: onPostCreate)
+    }
+    
+    private func onPostCreate(request: Request) async throws -> Response {
+        let speaker = try request.content.decode(Speaker.self)
+        let image = try request.content.decode(ImageUpload.self)
+        
+        let app = request.application
+        let filename = "\(image.profileImage.filename)-\(UUID.generateRandom().uuidString)"
+        let path = app.directory.publicDirectory + "img/\(filename)"
+        let handle = try app.fileio.openFile(path: path,
+                                             mode: .write,
+                                             flags: .allowFileCreation(posixMode: 0x744),
+                                             eventLoop: request.eventLoop).wait()
+        try app.fileio.write(fileHandle: handle, buffer: image.profileImage.data, eventLoop: request.eventLoop).wait()
```

Comment:
Can we use the async versions of these APIs ? I think theyre around now ( no big if not ) [docs](https://docs.vapor.codes/4.0/files/)